### PR TITLE
remove data_source.get_data slugs arg

### PR DESCRIPTION
**Pull request by dannyroberts**
As far as I can tell this functionality isn't used anywhere,
and having it around makes the code more confusing.
Also, many of the subclasses just ignore that arg

### CODE REVIEW

**Comment by dannyroberts** [comment]
@czue I wasn't 100% sure about this one, because there are other methods called `get_data` so it's hard to do a simple search and know you didn't make a mistake. I did search for regex `\.get_data\([^)]` (too find all method calls of something called "get_data" that aren't argless) and they all seemed to be some other `get_data` method.

If there are no calls to this (other than the one testing the API, which I changed), do you agree it's better to remove and reimplement (probably differently) if and when it's needed?

**Comment by czue** [comment]
I'm not sure either and don't actually know what this is for. I think @snopoke originally designed this. one thing you could do is `soft_assert` that `slugs` is always `None` and then delete it in a week or two.

**Comment by snopoke** [comment]
I don't think this is used. Fine to get rid of it.

**Comment by snopoke** [comment]
Tests should be fixed by https://github.com/dimagi/commcare-hq/pull/9275

**Comment by dannyroberts** [comment]
@czue `soft_assert` would be nice, but there are so many subclass method overrides that don't call super that I'd have to litter them throughout the code (which is possible but more annoying). I'd consider doing that, but @snopoke seems confident nothing uses it (which matches my own conclusion searching the code), so I'm tempted to just merge. Thoughts?

**Comment by snopoke** [comment]
I think pretty safe to merge.

On 25 November 2015 at 17:06, Daniel Roberts <notifications@github.com>
wrote:

> @czue <https://github.com/czue> soft_assert would be nice, but there are
> so many subclass method overrides that don't call super that I'd have to
> litter them throughout the code (which is possible but more annoying). I'd
> consider doing that, but @snopoke <https://github.com/snopoke> seems
> confident nothing uses it (which matches my own conclusion searching the
> code), so I'm tempted to just merge. Thoughts?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/9272#issuecomment-159634908>.
>



-- 
Simon Kelly
Senior Engineer | Dimagi South Africa

**Comment by dannyroberts** [comment]
:goberserk:

File: corehq/apps/reports/commtrack/data_sources.py [hunk 1]
```diff
@@ -243,7 +243,7 @@ def filter_by_program(self, stock_states):
             )
         )
 
-    def get_data(self, slugs=None):
+    def get_data(self):
```
Comment (kkrampa) [inline corehq/apps/reports/commtrack/data_sources.py]: @dannyroberts @snopoke @gcapalbo 
It seems slugs is still used here:
https://github.com/dimagi/commcare-hq/pull/9272/files#diff-7300848b9db6b7a41ec64acb711b91bfR275

Comment (snopoke) [inline corehq/apps/reports/commtrack/data_sources.py]: Can just remove the `slugs` param form `raw_product_states`

Comment (dannyroberts) [inline corehq/apps/reports/commtrack/data_sources.py]: Oops, thanks for catching and fixing this!

**Pull request merged**